File: JAVA/jzbus/src/test/java/net/zbus/pool/ReqRepClient.java
package net.zbus.pool;

import net.zbus.BusClient;
import net.zbus.ConnectionConfig;
import net.zbus.ZMsg;

public class ReqRepClient {

	public static void main(String[] args) {
		ConnectionConfig config = new ConnectionConfig();
		// config.setVerbose(true);
		BusClient client = new BusClient(config);

		ZMsg msg = new ZMsg();
		msg.pushBack("hello");
		msg = client.request("helloworld", "", msg, 2500);
		msg.dump();
		
		client.destroy();
	}

}

File: MS.NET/clrzbus/examples/reqrep/ReqRepClient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using zbus;
namespace zbus
{
    class ReqRepClient
    {
        public static void Main(string[] args)
        {
            ConnectionConfig config = new ConnectionConfig();
            config.Host = "127.0.0.1";
            config.Port = 15555;
            //config.Verbose = true;
         
            BusClient client = new BusClient(config);

            ZMsg msg = new ZMsg();
            msg.PushBack("hello");
            msg = client.Request("helloworld", "", msg);
            msg.Dump();

            client.Destroy();

            Console.ReadKey();
        } 
    }
}
